lua/entities/gmod_wire_expression2/core/cl_sound.lua

```diff
@@ -0,0 +1,367 @@
+local E2Sounds = {}
+local BlockedPlayers = {}
+
+local wire_expression2_sound_enabled = CreateConVar( "wire_expression2_sound_enabled_cl", 2, {FCVAR_ARCHIVE},"2: Anyone's sounds can be heard, 1: Only friend's sounds will be heard, 0: Only your own sounds will be heard")
+cvars.AddChangeCallback("wire_expression2_sound_enabled_cl", function(name, old, new)
+	if new~=2 then
+		for k,v in pairs(E2Sounds) do
+			v.SoundChannel:Stop()
+		end
+		E2Sounds = {}
+	end
+end)
+
+local function doPlayerBlocking(args, cb)
+	local name = args[1]:lower()
+	local players = E2Lib.filterList(player.GetAll(), function(ent) return ent:GetName():lower():match(name) end)
+	if #players == 1 then
+		cb(players[1])
+	elseif #players > 1 then
+		ply:PrintMessage( HUD_PRINTCONSOLE, "More than one player matches that name!" )
+	else
+		ply:PrintMessage( HUD_PRINTCONSOLE, "No player names found with " .. args[1] )
+	end
+end
+
+concommand.Add("wire_expression2_sound_blockplayer",function(ply,com,args)
+	doPlayerBlocking(args, function(found)
+		BlockedPlayers[found:SteamID()] = true
+		
+		for k,v in pairs(E2Sounds) do
+			if v.Player == found then
+				v.SoundChannel:Stop()
+				E2Sounds[k] = nil
+			end
+		end
+	end)
+end)
+
+concommand.Add("wire_expression2_sound_unblockplayer",function(ply,com,args)
+	doPlayerBlocking(args, function(found)
+		BlockedPlayers[found:SteamID()] = nil
+	end)
+end )
+
+local function moveSounds()
+	for k,v in pairs(E2Sounds) do
+		if v.SoundChannel then
```

Comment:
`// /* */ and !` I can understand, but `continue` is legitimately useful and I don't see a reason not to use it.

The argument that it doesn't work in _other environments_ is rather silly because 80% of wiremod won't work in another environment anyways.